animation_nodes/nodes/matrix/distribute_matrices.pyx

```diff
@@ -276,16 +286,16 @@ class DistributeMatricesNode(bpy.types.Node, AnimationNode):
         iSin = sin(startAngle)
         stepCos = cos(angleStep)
         stepSin = sin(angleStep)
-
+        
         for i in range(amount):
             f = <float>i * factor
 
             size = f * (endSize - startSize) + startSize
-            radius = f * (endRadius - startRadius) + startRadius
+            radius = radius = (endRadius - startRadius) * radiusInterpolation.evaluate(f) + startRadius
 
             position.x = iCos * radius
             position.y = iSin * radius
-            position.z = 0
+            position.z = spiralHeight * heightInterpolation.evaluate(f) - spiralHeight / 2 * self.centerSpiral
```

Comment:
Sorry for the miscommunication. I meant to inline only the step, the offset should be left as is outside the loop. With parentheses for clarity. 
```
(spiralHeight / 2) * self.centerSpiral
```